### PR TITLE
Possibly removing scope cloning

### PR DESCRIPTION
**Pull request by mlegenhausen**
This pull request improves the performance massivly when using `angular-google-maps` in more complex apps by removing all `_.clone(this.model, true)` statements. Which make a deep scope cloning. But the cloned scope is never used... In our app this clone process blocks the app for several seconds...

### CODE REVIEW

**Comment by mlegenhausen** [comment]
Ok I see the unit tests fail, but I was not able to execute them locally... maybe you can take a look? Is deep cloning really necessary?

**Comment by nmccready** [comment]
It's cloned for a reason to prevent recursive updates.

**Comment by nmccready** [comment]
To test locally:

`npm install`
`bower install`
'grunt curl`
`grunt default`

**Comment by nmccready** [comment]
We can debate this but you ned to respond.

**Comment by nmccready** [comment]
As you can see it is not worthless as it is used for pieceMeal https://github.com/angular-ui/angular-google-maps/blob/master/src/coffee/directives/api/utils/models-watcher.coffee#L25

File: src/coffee/directives/api/models/child/window-child-model.coffee [hunk 1]
```diff
@@ -9,10 +9,6 @@ angular.module('uiGmapgoogle-maps.directives.api.models.child')
         constructor: (@model, @scope, @opts, @isIconVisibleOnClick, @mapCtrl, @markerScope, @element,
           @needToManualDestroy = false, @markerIsVisibleAfterWindowClose = true) ->
 
-          #where @model is a reference to model in the controller scope
-          #clonedModel is a copy for comparison
```
Comment (nmccready) [inline src/coffee/directives/api/models/child/window-child-model.coffee]: As this comment says this is used for comparison which is done in model-watcher

**Comment by nmccready** [comment]
Removing this will screw up updates; a work around would be to make cloning optional and force doRebuildAll thus not doing updating. This will cause "flash" updating where the entire array will be erased and redrawn.

**Comment by nmccready** [comment]
Closing until there is some feedback.

**Comment by mlegenhausen** [comment]
Sorry was ill for the last week.

I think I understand. We have the same problems, but cause we leverage immutable data structures in our angular app, we do not need to clone everything to detect changes in an async way. But of cause this will not work for you. But it would be nice if this mode could be disabled, cause the app keeps care about not modifying the data.

**Comment by nmccready** [comment]
Fix child code do the checks on cloning. Where you might want to make it a function.

(marker-child-model)[https://github.com/angular-ui/angular-google-maps/blob/master/src/coffee/directives/api/models/child/marker-child-model.coffee#L26]

To:
```coffee
maybeCloneModel: =>
  @clonedModel = _.clone(@model,true) if not @scope.doRebuildAll
```

Finally use doRebuild all in your implementation markup `<markers doRebuildAll="true"></markers>`.

**Comment by nmccready** [comment]
Either I can reopen this PR and you can max the appropriate fixes (w rebase). Otherwise close the PR and make an issue linking this and then make a new PR.

**Comment by mlegenhausen** [comment]
Ok thanks.

**Comment by nmccready** [comment]
So should I reopen and continue changes here or new ticket?

**Comment by mlegenhausen** [comment]
Don't reopen, when then a new ticket.

**Pull request closed without merging**